test/jdk/javax/swing/JTableHeader/TableHeaderBorderPositionTest.java

```diff
@@ -0,0 +1,128 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.imageio.ImageIO;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.JTableHeader;
+import javax.swing.UIManager;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+import java.io.File;
+import java.io.IOException;
+
+/*
+ * @test
+ * @bug 8311031
+ * @summary Test to validate JTable header border vertical line is
+ *          aligned with data grid lines (Metal L&F).
+ * @run main TableHeaderBorderPositionTest
+ */
+
+public class TableHeaderBorderPositionTest {
+    private static final int WIDTH = 300;
+    private static final int HEIGHT = 150;
+    private static final double SCALE = 2.25;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        SwingUtilities.invokeAndWait(TableHeaderBorderPositionTest::test);
+    }
+
+    private static void test() {
+        String[][] data = {
+                {"1", "1"}
+        };
+
+        String[] columnNames = {"Size", "Size"};
+        JTable table = new JTable(data, columnNames);
+        table.setSize(WIDTH, HEIGHT);
+
+        final JTableHeader header = table.getTableHeader();
+        Dimension size = header.getPreferredSize();
+        header.setSize(size);
+
+        int w = (int)Math.ceil(SCALE * size.width);
+        int h = (int)Math.ceil(SCALE * size.height);
+
+        BufferedImage imgHeader = new BufferedImage(w, h,
+                BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2d = imgHeader.createGraphics();
+        g2d.scale(SCALE, SCALE);
+        try {
+            header.paint(g2d);
+        } finally {
+            g2d.dispose();
+        }
+
+        w = (int)Math.ceil(SCALE * WIDTH);
+        h = (int)Math.ceil(SCALE * HEIGHT);
+        BufferedImage imgData = new BufferedImage(w, h,
+                BufferedImage.TYPE_INT_RGB);
+        g2d = imgData.createGraphics();
+        g2d.scale(SCALE, SCALE);
+        try {
+            table.paint(g2d);
+        } finally {
+            g2d.dispose();
+        }
+
+        int verticalLineCol = (int)(table.getTableHeader().
+                getColumnModel().getColumn(0).getWidth() * SCALE) - 2;
+        int expectedRGB = imgData.getRGB(verticalLineCol, 0);
+
+        for (int y = 0; y < imgHeader.getHeight(); y++) {
+            for (int x = verticalLineCol; x < verticalLineCol + 3; x++) {
+                if (expectedRGB != imgHeader.getRGB(x, y)) {
+                    saveImage(imgHeader, "FailureImageHeader.png");
+                    throw new RuntimeException("Header Comparison Test " +
+                            "Failed at <" + x + ", " + y + ">");
+                }
+            }
+        }
+
+        int maxVerticalLine = (int) Math.ceil(
+                table.getRowCount() * table.getRowHeight() * SCALE);
```

Comment:
```suggestion
        int maxVerticalLine = (int)Math.ceil(
                table.getRowCount() * table.getRowHeight() * SCALE);
```
In all the other cases, you didn't add a space after the cast operator.